src/VstsSyncMigrator.Core/Execution/MigrationContext/WorkItemMigrationContext.cs

```diff
@@ -705,5 +712,26 @@ private WorkItemData ReplayRevisions(List<RevisionItem> revisionsToMigrate, Work
 
             return targetWorkItem;
         }
+
+        private static bool RevisionHasValidIterationPath(WorkItemData targetWorkItem)
+        {
+            var isValid = true;
+            var workItem = (WorkItem)targetWorkItem.internalObject;
+            var fails = workItem.Validate();
+
+            if (fails.Count > 0)
+            {
+                foreach (Field f in fails)
+                {
+                    // We cannot save a revision when it has no IterationPath
+                    if (f.ReferenceName == "System.IterationPath")
+                    {
+                        isValid = false;
```

Comment:
```suggestion
                        return false;
```

If you do this and change the last line too with the other suggestion, you can remove the `isValid` var. It's a small thing but it does make it slightly faster because then you quit on the first invalid field and don't bother checking others.